pyccel/ast/internals.py

```diff
@@ -292,8 +293,10 @@ def max_precision(objs : list, dtype = None, allow_native = True):
         return max(def_prec if o.precision == -1 \
                 else o.precision for o in objs if o.dtype is dtype)
     else:
-        return max(default_precision[str(o.dtype)] if o.precision == -1 \
-                else o.precision for o in objs)
+        ndarray_list = [o for o in objs if getattr(o, 'is_ndarray', False)]
+        if ndarray_list:
+            return get_final_precision(max(ndarray_list, key=attrgetter('precision')))
```

Comment:
Is there a reason to prefer `attrgetter` to ` get_final_precision(max(ndarray_list, key=lambda l: l.precision))`?

Reply:
I guess using the lambda function improves the readability a bit, i'll switch to that i guess.

Reply:
I don't know if it's more readable, but it removes an unnecessary import

Reply:
Actually it looks like your choice is faster: https://stackoverflow.com/a/54061395/5957195 :+1:

Reply:
Alright i'll leave it as it is.